source/s3_client.c

```diff
@@ -1130,6 +1189,55 @@ static void s_s3_client_schedule_process_work_synced_default(struct aws_s3_clien
     client->synced_data.process_work_task_scheduled = true;
 }
 
+/* Task function for trying to find a request that can be processed. */
+static void s_s3_client_trim_buffer_pool_task(struct aws_task *task, void *arg, enum aws_task_status task_status) {
+    AWS_PRECONDITION(task);
+    (void)task;
+    (void)task_status;
+
+    struct aws_s3_client *client = arg;
+    AWS_PRECONDITION(client);
+
+    aws_s3_client_lock_synced_data(client);
+    client->synced_data.trim_buffer_pool_task_scheduled = false;
+    aws_s3_client_unlock_synced_data(client);
```

Comment:
we could move `trim_buffer_pool_task_scheduled` from `synced_data` to `threaded_data` and not bother with locks here, right? we only ever schedule this and run it from the client-work-thread?